client/src/partials/debtor_groups/debtor_groups.js

```diff
@@ -0,0 +1,154 @@
+angular.module('bhima.controllers')
+.controller('DebtorGroupsController', DebtorGroupsController);
+
+DebtorGroupsController.$inject = ['$http', '$translate', 'DebtorGroupService', 'SessionService', 'uuid'];
+
+/**
+ * Debtor Groups Controller
+ * This controller is about the debtor groups management module in the admin zone
+ * It's responsible for creating and updating a debtor group
+ */
+function DebtorGroupsController($http, $translate, DebtorGroup, Session, uuid) {
+  'use strict';
+
+  var vm = this;
+
+  /** breadcrumb configurations */
+  vm.bcPaths = [
+    {
+      label: $translate.instant('DEBTOR_GRP.TITLE'),
+      current: true
+    }
+  ];
+  vm.bcButtons = [
+    {
+      dataMethod : 'create',
+      color  : 'btn-default',
+      icon   : 'glyphicon glyphicon-plus-sign',
+      label  : $translate.instant('DEBTOR_GRP.NEW'),
+      action : create
+    }
+  ];
+
+  /** init variables */
+  vm.state = {};
+  vm.debtorGroup = {};
+
+  /** Expose to the view */
+  vm.update = update;
+  vm.cancel = cancel;
+  vm.submit = submit;
+  vm.refreshValidation = refreshValidation;
+
+  /** Load debtor groups */
+  debtorGroupsList()
+  .then(accountList)
+  .then(priceList)
+  .catch(handler);
+
+  function debtorGroupsList() {
+    return DebtorGroup.read()
+    .then(function (list) {
+      vm.debtorGroupList = list;
+    });
+  }
+
+  /** @fixme Need service for getting accounts list */
+  function accountList () {
+    return $http.get('/accounts')
+    .then(function (accounts) {
+      vm.accounts = accounts;
+    });
+  }
+
+  /** @fixme Need service for getting prices list */
+  function priceList () {
+    return $http.get('/prices')
+    .then(function (prices) {
+      vm.prices = prices;
+    });
+  }
+
+  function update(uuid) {
+    initialise('action', 'update', uuid);
+  }
+
+  function create() {
+    initialise('action', 'create');
+  }
+
+  function cancel() {
+    initialise('default');
+  }
+
+  function submit(invalid) {
+    if (invalid) {
+      vm.state.errored = true;
+    }
+    if (vm.debtorGroup.name && vm.action === 'create') {
```

Comment:
Do you need to check `vm.debtorGroup.name`?  Isn't it a required field?  Angular's form validation should mark it invalid and you can return if the form is invalid.